include/swupd.h

```diff
@@ -84,6 +84,7 @@ struct file {
 	int last_change;
 	struct update_stat stat;
 
+	unsigned int fd_valid : 1;
```

Comment:
@pohly @rojkov Any thoughts on this approach? I really think fd initialized to -1 would be better than using the separate fd_valid field.